File: scala/010/Main.scala
/*
 * ProjectEuler 010
 * Scala / Main.scala
 * copyright (c) 2015 Susisu
 *
 * Scala 2.11.4
 * `scalac Main.scala -d bin`
 */

import math._

object Main {
    def main(args: Array[String]) {
        val n = 2000000;
        val table = new Array[Boolean](n);
        table(0) = false;
        table(1) = false;
        for (i <- 2 until n) {
            table(i) = true;
        }

        val r = floor(sqrt(n)).toInt;
        for (i <- 2 to r) {
            if (table(i)) {
                for (j <- i * i until n by i) {
                    table(j) = false;
                }
            }
        }

        var total: Long = 0;
        for (i <- 0 until n) {
            if (table(i)) {
                total += i;
            }
        }
        println(total);
    }
}

File: java/010/Main.java
/*
 * ProjectEuler 010
 * Java / Main.java
 * copyright (c) 2015 Susisu
 *
 * Java SE 1.8.0_25
 * `javac Main.java -d bin`
 */

public class Main {
    private static final int N = 2000000;

    public static void main(String[] args) {
        boolean table[] = new boolean[N];
        table[0] = false;
        table[1] = false;
        for (int i = 2; i < N; i++) {
            table[i] = true;
        }

        int r = (int)Math.floor(Math.sqrt(N));
        for (int i = 2; i <= r; i++) {
            if (table[i]) {
                for (int j = i * i; j < N; j += i) {
                    table[j] = false;
                }
            }
        }

        long total = 0;
        for (int i = 0; i < N; i++) {
            if (table[i]) {
                total += i;
            }
        }
        System.out.println(total);
    }
}
